[Bug] fresh anonymous named variable crashes with NoSuchElementException

File:
regression-new/withConfig2/test.k
```
module TEST

   imports INT

   configuration <x>  <k> $PGM:K </k> </x>

   syntax KItem ::= freshIntx() [function]

   rule freshIntx() => !_I:Int

   rule 5 => freshIntx()

   rule 6 => test()

   syntax Int ::= test() [function]

   rule [[ test() => 7 ]]
        <k> 6 ...</k>

endmodule
```
Crashes with:
```
regression-new/withConfig2$ kompile --syntax-module TEST --backend llvm --debug test.k
org.kframework.utils.errorsystem.KEMException: [Error] Internal: Could not find binding for variable while compiling pattern matching. (NoSuchElementException: key not found: Var'BangUnds'I)
	Source(/home/radu/work/k/k-distribution/tests/regression-new/withConfig2/./test.k)
	Location(9,9,9,31)
	at org.kframework.utils.errorsystem.KEMException.create(KEMException.java:130)
	at org.kframework.utils.errorsystem.KEMException.internalError(KEMException.java:72)
	at org.kframework.backend.llvm.matching.Matrix.getLeaf(Matrix.scala:564)
	at org.kframework.backend.llvm.matching.Matrix.compileInternal(Matrix.scala:646)
	at org.kframework.backend.llvm.matching.Matrix.compile(Matrix.scala:615)
	at org.kframework.backend.llvm.matching.Generator$.mkDecisionTree(Generator.scala:145)
	at org.kframework.backend.llvm.matching.Matching$.$anonfun$writeDecisionTreeToFile$5(Matching.scala:46)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:234)
	at scala.collection.Iterator.foreach(Iterator.scala:929)
	at scala.collection.Iterator.foreach$(Iterator.scala:929)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1417)
	at scala.collection.IterableLike.foreach(IterableLike.scala:71)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:70)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at scala.collection.TraversableLike.map(TraversableLike.scala:234)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:227)
	at scala.collection.AbstractTraversable.map(Traversable.scala:104)
	at org.kframework.backend.llvm.matching.Matching$.writeDecisionTreeToFile(Matching.scala:39)
	at org.kframework.backend.llvm.matching.Matching.writeDecisionTreeToFile(Matching.scala)
	at org.kframework.backend.llvm.LLVMBackend.accept(LLVMBackend.java:50)
	at org.kframework.kompile.KompileFrontEnd.run(KompileFrontEnd.java:75)
	at org.kframework.main.FrontEnd.main(FrontEnd.java:62)
	at org.kframework.main.Main.runApplication(Main.java:118)
	at org.kframework.main.Main.runApplication(Main.java:108)
	at org.kframework.main.Main.main(Main.java:56)
Caused by: java.util.NoSuchElementException: key not found: Var'BangUnds'I
	at scala.collection.MapLike.default(MapLike.scala:232)
	at scala.collection.MapLike.default$(MapLike.scala:231)
	at scala.collection.AbstractMap.default(Map.scala:59)
	at scala.collection.MapLike.apply(MapLike.scala:141)
	at scala.collection.MapLike.apply$(MapLike.scala:140)
	at scala.collection.AbstractMap.apply(Map.scala:59)
	at org.kframework.backend.llvm.matching.Matrix.$anonfun$getLeaf$10(Matrix.scala:562)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:234)
	at scala.collection.immutable.List.foreach(List.scala:389)
	at scala.collection.TraversableLike.map(TraversableLike.scala:234)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:227)
	at scala.collection.immutable.List.map(List.scala:295)
	at org.kframework.backend.llvm.matching.Matrix.getLeaf(Matrix.scala:562)
	... 22 more
```
Removing the `_` gives out a warning:
```
regression-new/withConfig2$ kompile --syntax-module TEST --backend llvm --debug test.k
[Warning] Compiler: Variable '!I' defined but not used. Prefix variable name
with underscore if this is intentional.
	Source(/home/radu/work/k/k-distribution/tests/regression-new/withConfig2/./test.k)
	Location(9,24,9,26)
```

--- a/k-distribution/tests/regression-new/withConfig2/test.k
+++ b/k-distribution/tests/regression-new/withConfig2/test.k
@@ -6,7 +6,7 @@ module TEST
 
    syntax KItem ::= freshIntx() [function]
 
-   rule freshIntx() => !I:Int
+   rule freshIntx() => !_I:Int
 
    rule 5 => freshIntx()
 

